FDK/Tools/SharedData/FDKScripts/CheckOutlinesUFO.py

```diff
@@ -12,1020 +13,1145 @@
+        g2 = expand_names(glyph_range[1])
+        glyph_name = "%s-%s" % (g1, g2)
+
+    elif glyph_name[0] == "/":
+        glyph_name = "cid" + glyph_name[1:].zfill(5)
+        if glyph_name == "cid00000":
+            glyph_name = ".notdef"
+            nameAliasDict[glyph_name] = "cid00000"
+
+    elif glyph_name.startswith("cid") and (len(glyph_name) < 8):
+        glyph_name = "cid" + glyph_name[3:].zfill(5)
+        if glyph_name == "cid00000":
+            glyph_name = ".notdef"
+            nameAliasDict[glyph_name] = "cid00000"
+
+    return glyph_name
+
+
+def get_glyph_names(glyph_tag, font_glyph_list, font_file_name):
+    glyph_name_list = []
+    range_list = glyph_tag.split("-")
+    prev_gid = get_glyph_id(range_list[0], font_glyph_list)
+    if prev_gid is None:
+        if len(range_list) > 1:
+            print("\tWarning: glyph ID <%s> in range %s from glyph selection "
+                  "list option is not in font. <%s>." %
+                  (range_list[0], glyph_tag, font_file_name))
+        else:
+            print("\tWarning: glyph ID <%s> from glyph selection list option "
+                  "is not in font. <%s>." % (range_list[0], font_file_name))
+        return None
+    try:
+        glyph_name = font_glyph_list[prev_gid]
+    except IndexError:
+        print("\tWarning: glyph ID <%s> from glyph selection list option is "
+              "not in font. <%s>." % (prev_gid, font_file_name))
+        return None
+    glyph_name_list.append(glyph_name)
+
+    for glyphTag2 in range_list[1:]:
+        gid = get_glyph_id(glyphTag2, font_glyph_list)
+        if gid is None:
+            print("\tWarning: glyph ID <%s> in range %s from glyph selection "
+                  "list option is not in font. <%s>." %
+                  (glyphTag2, glyph_tag, font_file_name))
+            return None
+        for i in range(prev_gid + 1, gid + 1):
+            glyph_name_list.append(font_glyph_list[i])
+        prev_gid = gid
+
+    return glyph_name_list
+
+
+def filter_glyph_list(options, font_glyph_list, font_file_name):
+    # Return the list of glyphs which are in the intersection of the argument
+    # list and the glyphs in the font.
+    # Complain about glyphs in the argument list which are not in the font.
+    if not options.glyphList:
+        glyph_list = font_glyph_list
+    else:
+        # expand ranges:
+        glyph_list = []
+        for glyphTag in options.glyphList:
+            glyph_names = \
+                get_glyph_names(glyphTag, font_glyph_list, font_file_name)
+            if glyph_names is not None:
+                glyph_list.extend(glyph_names)
+    return glyph_list
+
+
+def get_digest(d_glyph):
+    """copied from robofab ObjectsBase.py.
+    """
+    mp = DigestPointPen()
+    d_glyph.drawPoints(mp)
+    digest = list(mp.getDigestPointsOnly(needSort=False))
+    digest.append((len(d_glyph.contours), 0))
+    return digest
+
+
+# noinspection PyProtectedMember
```

Comment:
what is this comment about?

Reply:
PyCharm has an inspection that complains about accessing contour._points[] because _points is a protected member of the class (based on the leading underscore). Since that seemed to me to be unavoidable in this code, I added that comment to suppress the inspection for this function.

Reply:
ok